src/runtime/micro/host_low_level_device.cc

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file host_low_level_device.cc
+ * \brief emulated low-level micro device implementation on host machine
+ */
+
+#include <sys/mman.h>
+#include <cstring>
+#include "low_level_device.h"
+#include "micro_common.h"
+
+namespace tvm {
+namespace runtime {
+/*!
+ * \brief emulated low-level device on host machine
+ */
+class HostLowLevelDevice final : public LowLevelDevice {
+ public:
+  /*!
+   * \brief constructor to initialize on-host memory region to act as device
+   * \param num_bytes size of the emulated on-device memory region
+   */
+  explicit HostLowLevelDevice(size_t num_bytes)
+    : size_(num_bytes) {
+    size_t size_in_pages = (num_bytes + kPageSize - 1) / kPageSize;
+    int mmap_prot = PROT_READ | PROT_WRITE | PROT_EXEC;
```

Comment:
Note that the entire HostLowLevelDevice memory is executable. Do we want to change this to only be the code section?

Reply:
I think it would definitely be helpful to apply the appropriate permissions to every section.  Personally, I'd like to add a `TODO` and implement it in a future PR, but if anyone has strong opinions on it, we can implement that in the current PR.

Reply:
I think it's fine for now, and simplifies testing.